frontend/src/pages/Assignments/CreateAssignment/index.tsx

```diff
@@ -63,7 +63,7 @@ const CreateAssignment: React.FC = () => {
 
     const handleSubmit = async (data: IAssignmentFormData) => {
         await createAssignmentTemplate(data.classroomId, data.templateRepo!);
-        await createAssignment(data.templateRepo!.id, data);
+        await createAssignment(data.templateRepo!, data);
```

Comment:
I'd like a more robust way to handle a null value here rather than throwing. non-null assertions are usually indicative of a bad design decision